webpack/JobWizard/steps/HostsAndInputs/index.js

```diff
@@ -1,66 +1,159 @@
+  hostsController,
+  hostQuerySearchID,
+} from '../../JobWizardConstants';
 import { WizardTitle } from '../form/WizardTitle';
+import { HostSelect } from './HostSelect';
+import { HostSearch } from './HostSearch';
 
 const HostsAndInputs = ({
   templateValues,
   setTemplateValues,
-  selectedHosts,
-  setSelectedHosts,
+  selected,
+  setSelected,
+  hostsSearchQuery,
+  setHostsSearchQuery,
 }) => {
+  const [hostMethod, setHostMethod] = useState(hostMethods.hosts);
   const templateInputs = useSelector(selectTemplateInputs);
-  const hostMethods = [
-    __('Hosts'),
-    __('Host collection'),
-    __('Host group'),
-    __('Search query'),
-  ];
-  const [hostMethod, setHostMethod] = useState(hostMethods[0]);
+  const withKatello = useSelector(selectWithKatello);
+  const dispatch = useDispatch();
+
+  const selectedHosts = selected.hosts;
+  const setSelectedHosts = newSelected =>
+    setSelected(prevSelected => ({
+      ...prevSelected,
+      hosts: newSelected(prevSelected.hosts),
+    }));
+  const selectedHostCollections = selected.hostCollections;
+  const setSelectedHostCollections = newSelected =>
+    setSelected(prevSelected => ({
+      ...prevSelected,
+      hostCollections: newSelected(prevSelected.hostCollections),
+    }));
+  const selectedHostGroups = selected.hostGroups;
+  const setSelectedHostGroups = newSelected => {
+    setSelected(prevSelected => ({
+      ...prevSelected,
+      hostGroups: newSelected(prevSelected.hostGroups),
+    }));
+  };
+
+  const clearSearch = () => {
+    dispatch(resetData(hostsController, hostQuerySearchID));
+    setHostsSearchQuery('');
+  };
   return (
-    <>
+    <div className="target-hosts-and-inputs">
       <WizardTitle title={WIZARD_TITLES.hostsAndInputs} />
       <Form>
         <FormGroup fieldId="host_selection">
-          <SelectField
-            fieldId="host_methods"
-            options={hostMethods}
-            setValue={setHostMethod}
-            value={hostMethod}
-          />
-          <SelectedChips
-            selected={selectedHosts}
-            setSelected={setSelectedHosts}
-          />
+          <InputGroup>
+            <SelectField
+              className="target-method-select"
+              toggleIcon={<FilterIcon />}
+              fieldId="host_methods"
+              options={Object.values(hostMethods)}
+              setValue={setHostMethod}
+              value={hostMethod}
+            />
+            {hostMethod === hostMethods.searchQuery && (
+              <HostSearch
+                setValue={setHostsSearchQuery}
+                value={hostsSearchQuery}
+              />
+            )}
+            {hostMethod === hostMethods.hosts && (
+              <HostSelect
+                selectedHosts={selectedHosts}
+                setSelectedHosts={setSelectedHosts}
+                apiKey={HOSTS}
+                url="/api/hosts?per_page=100"
+                placeholderText={__('Filter by hosts')}
+              />
```

Comment:
The whole host fetching is quite resource intensive, considering we only need total count and a hostname for each of the hosts. Could we maybe use graphql here?